README.md
<img src='./images/banner.png' />

>Mobile App for Android and iOS designed to monitor WoW server https://elysium-project.org/. It makes the process of checking elysium-project easier while on the go.

* **Current Status**: Alpha

#### Status
<img src='./images/android.png' height='400' /> <img src='./images/ios.png' height='400' />

#### Realm Stats
<img src='./images/android_realm_stats.png' height='400' /> <img src='./images/ios_realm_stats.png' height='400' />
#### Realm Alert Push Notifications
<img src='./images/android_push.png' height='400' />

#### Download

###### Android (Alpha)
https://play.google.com/store/apps/details?id=com.elysiumstatus

###### iOS (Waiting on Developer Account)
Not yet released

* **
# Getting Started
These instructions will get you a copy of the project up and running on your local machine for development and testing purposes.

## Prerequisites
What things you need to install the project
* [React-Native](https://facebook.github.io/react-native)

React native has a great setup guide which should guide you through everything you need to do get started
https://facebook.github.io/react-native/docs/getting-started.html

#### Config files (required)
#### Android (google-services.json)
1. Visit https://firebase.google.com/ and create a free account.
2. Create a new project
3. Click Add Firebase to your Android app
4. Use `com.elysiumstatus.dev` as Package name
5. Add a nickname something like `Elysium Status Development`
6. Click add app
7. It should have downloaded a `google-services.json` file.
8. Move that file to `android/app/` directory
9. That's it!

## Installing
After you have react-native running, forked & cloned the repo. Running the development project is simple.
### Android
In your project directory use the [`react-native-cli`](https://www.npmjs.com/package/react-native-cli) to start the project
```
react-native run-android
```
It should spawn a react native packager and the development app should automatically be installed. The cli will attempt to start the app on the emulator, but will not find it. It will need to be manually opened.

>[*Make sure you have a device connected or an emulator running*](https://facebook.github.io/react-native/docs/getting-started.html#starting-the-android-virtual-device)

### iOS (macOS Only)
In your project directory use the [`react-native-cli`](https://www.npmjs.com/package/react-native-cli) to start the project
```
react-native run-ios
```
It should spawn a react native packager and the simulator should start and launch the development app.

## Tests
TODO: Write tests using [jest](https://facebook.github.io/jest/)

## Built with
* [ReactNative](https://facebook.github.io/react-native/) - Native App Framework
* [React](https://facebook.github.io/react/) - Javascript UI Library
* [Redux](http://redux.js.org/) - State container for JavaScript
* [Material Design](https://material.io/guidelines/) - Design Guidelines
* [react-native-vector-icons](https://github.com/oblador/react-native-vector-icons) - Vector icons for react native

## Getting involved
Full contribution guidelines listed in [CONTRIBUTING.md](CONTRIBUTING.md)

### Bug Reports
Please submit all bug reports through the repository's Issue Tracker and they will be tagged and prioritized.

### Product Roadmap
List of features where help is desired.

1. ~~Realm Stats - Enhanced realm Stats from https://elysium-project.org/status~~
2. Push Notifications - Realm status notifications
  * User opt in
  * Customizable Notifications (UP, Down)
3. Server Status History - Detailed history of Server statuses
4.  Autologin - Login into wow from anywhere (Bypass the Queues)

## Versioning
We use [SemVer](http://semver.org/) for versioning.

## Authors
* [emmoistner](https://github.com/emmoistner) - *Initial work*

## License
This project is licensed under the MIT License - see the [LICENSE.md](LICENSE.md) file for details

# Acknowledgments
* **Private WoW Server Community**: Elysium Project, Nostalrius
* **Inspiration**: [Elysium-Status](https://github.com/ErlendEllingsen/Elysium-Status) - http://www.elysiumstatus.com


CHANGELOG.md
# Change Log

## [1.0.0-rc4](https://github.com/vanillastatus/VanillaStatusApp/tree/HEAD) (9/28/17)
**New Features**
- Realm Alerts (Beta) [\#26](https://github.com/vanillastatus/VanillaStatusApp/pull/36) ([code](https://github.com/vanillastatus/VanillaStatusApp/pull/36))
- New Api [\#37](https://github.com/vanillastatus/VanillaStatusApp/pull/37) ([code](https://github.com/vanillastatus/VanillaStatusApp/pull/37))

**Fixed Bugs**
- \[fix\] Add error screen [\#37](https://github.com/vanillastatus/VanillaStatusApp/pull/37) ([emmoistner](https://github.com/emmoistner))
- \[fix\] Handle Api Errors better [\#37](https://github.com/vanillastatus/VanillaStatusApp/pull/37) ([emmoistner](https://github.com/emmoistner))

**Merged pull requests:**

- \[feature\] Realm Alerts [\#37](https://github.com/vanillastatus/VanillaStatusApp/pull/37) ([emmoistner](https://github.com/emmoistner))
- \[fix\] New Api [\#36](https://github.com/vanillastatus/VanillaStatusApp/pull/36) ([emmoistner](https://github.com/emmoistner))

## [1.0.0-rc3](https://github.com/vanillastatus/VanillaStatusApp/tree/HEAD) (1/23/17)

**New Features**
- Realm Stats View [\#26](https://github.com/vanillastatus/VanillaStatusApp/issues/26) ([code](https://github.com/vanillastatus/VanillaStatusApp/pull/32))

**Fixed bugs:**

- Android Permission Issues [\#29](https://github.com/vanillastatus/VanillaStatusApp/issues/29)
- \[fix\] remove unused android permissions [\#30](https://github.com/vanillastatus/VanillaStatusApp/pull/30) ([emmoistner](https://github.com/emmoistner))
- \[fix\] add function to handle device rotation [\#23](https://github.com/vanillastatus/VanillaStatusApp/pull/23) ([emmoistner](https://github.com/emmoistner))
- \[fix\] ellipse title, subtitle material box closes \#18 [\#20](https://github.com/vanillastatus/VanillaStatusApp/pull/20) ([emmoistner](https://github.com/emmoistner))

**Closed issues:**

- Adjust box sizes for login & website. [\#27](https://github.com/vanillastatus/VanillaStatusApp/issues/27)
- Realm Stats Views [\#26](https://github.com/vanillastatus/VanillaStatusApp/issues/26)
- iOS App Icon [\#17](https://github.com/vanillastatus/VanillaStatusApp/issues/17)
- Android App Icon [\#16](https://github.com/vanillastatus/VanillaStatusApp/issues/16)
- Android Release Process [\#14](https://github.com/vanillastatus/VanillaStatusApp/issues/14)
- Grid view instead of card view. [\#5](https://github.com/vanillastatus/VanillaStatusApp/issues/5)
- First Release [\#4](https://github.com/vanillastatus/VanillaStatusApp/issues/4)
- Add product roadmap to readme [\#3](https://github.com/vanillastatus/VanillaStatusApp/issues/3)
- Add setup guide to readme [\#2](https://github.com/vanillastatus/VanillaStatusApp/issues/2)

**Merged pull requests:**

- \[feature\] realm stats [\#32](https://github.com/vanillastatus/VanillaStatusApp/pull/32) ([emmoistner](https://github.com/emmoistner))
- \[feature\] allow login server & website to group together [\#28](https://github.com/vanillastatus/VanillaStatusApp/pull/28) ([emmoistner](https://github.com/emmoistner))
- \[readme\] update readme with android link [\#25](https://github.com/vanillastatus/VanillaStatusApp/pull/25) ([emmoistner](https://github.com/emmoistner))
- \[feature\] add high res banner [\#22](https://github.com/vanillastatus/VanillaStatusApp/pull/22) ([emmoistner](https://github.com/emmoistner))
- \[feature\] App Icons [\#19](https://github.com/vanillastatus/VanillaStatusApp/pull/19) ([emmoistner](https://github.com/emmoistner))
- \[style\] changed primaryColor for taskbar, cleaned App name [\#11](https://github.com/vanillastatus/VanillaStatusApp/pull/11) ([emmoistner](https://github.com/emmoistner))
- \[bug\] change parser to handle new autoqueue format [\#10](https://github.com/vanillastatus/VanillaStatusApp/pull/10) ([emmoistner](https://github.com/emmoistner))
- \[fix\] loading background color now uses primary color [\#9](https://github.com/vanillastatus/VanillaStatusApp/pull/9) ([emmoistner](https://github.com/emmoistner))
- \[readme\] enhanced readme.md [\#8](https://github.com/vanillastatus/VanillaStatusApp/pull/8) ([emmoistner](https://github.com/emmoistner))
- \[feature\] Gridview closes \#5 [\#6](https://github.com/vanillastatus/VanillaStatusApp/pull/6) ([emmoistner](https://github.com/emmoistner))
- update readme [\#1](https://github.com/vanillastatus/VanillaStatusApp/pull/1) ([emmoistner](https://github.com/emmoistner))



\* *This Change Log was automatically generated by [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator)*


app/components/detail.js
import React, { PropTypes, Component } from 'react'
import {
  View,
  Text,
  ScrollView,
  Image,
  Animated,
  StyleSheet,
  StatusBar,
  TouchableOpacity,
  TouchableNativeFeedback,
  Platform
} from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'
import { Actions } from 'react-native-router-flux'

import DetailContent from './detail_content'

import { THEME } from '../config'
const { PRIMARY_COLOR } = THEME

const STATUS_BAR_HEIGHT = Platform.OS === 'android' ? StatusBar.currentHeight : 20
const NAVBAR_HEIGHT = 56
const HEADER_MAX_HEIGHT = 350
const HEADER_MIN_HEIGHT = STATUS_BAR_HEIGHT + NAVBAR_HEIGHT
const HEADER_SCROLL_DISTANCE =  HEADER_MAX_HEIGHT - HEADER_MIN_HEIGHT

class Detail extends Component {
  constructor(props) {
    super(props)

    this.state ={
      scrollY: new Animated.Value(0)
    }
  }

  renderBackButton() {
    return Platform.select({
      ios: () => (
        <TouchableOpacity
          hitSlop={{ top: 24, bottom: 24, left: 24, right: 24 }}
          onPress={() => Actions.pop()}
        >
          <View style={styles.navbarIcon}>
            <Icon name='arrow-back' size={24} color={'#ffffff'} />
          </View>
        </TouchableOpacity>
      ),
      android: () => (
        <TouchableNativeFeedback
          hitSlop={{ top: 24, bottom: 24, left: 24, right: 24 }}
          background={TouchableNativeFeedback.Ripple('#ffffff', true)}
          onPress={() => Actions.pop()}
        >
          <View style={styles.navbarIcon}>
            <Icon name='arrow-back' size={24} color={'#ffffff'} />
          </View>
        </TouchableNativeFeedback>
      )
    })()
  }

  renderContent() {
    return (
      <View style={{ paddingTop: HEADER_MAX_HEIGHT }}>
        <DetailContent {...this.props} />
      </View>
    )
  }

  render() {
    const headerHeight = this.state.scrollY.interpolate({
      inputRange: [ 0, HEADER_SCROLL_DISTANCE ],
      outputRange: [ HEADER_MAX_HEIGHT, HEADER_MIN_HEIGHT ],
      extrapolate: 'clamp'
    })

    const imageOpacity = this.state.scrollY.interpolate({
      inputRange: [ 0, HEADER_SCROLL_DISTANCE / 2, HEADER_SCROLL_DISTANCE ],
      outputRange: [ 1, 1, 0 ],
      extrapolate: 'clamp'
    })

    const imageTranslate = this.state.scrollY.interpolate({
      inputRange: [ 0, HEADER_SCROLL_DISTANCE ],
      outputRange: [ 0, -50 ],
      extrapolate: 'clamp'
    })

    const fontSizeTranslate = this.state.scrollY.interpolate({
      inputRange: [ 0, HEADER_SCROLL_DISTANCE / 2, HEADER_SCROLL_DISTANCE ],
      outputRange: [ 44, 44, 20 ],
      extrapolate: 'clamp'
    })

    const padLeftTranslate = this.state.scrollY.interpolate({
      inputRange: [ 0, HEADER_SCROLL_DISTANCE / 2, HEADER_SCROLL_DISTANCE ],
      outputRange: [ 16, 16, 72 ],
      extrapolate: 'clamp'
    })

    return (
      <View style={{ flex: 1, backgroundColor: PRIMARY_COLOR }}>
        <Animated.View
          pointerEvents='none'
          style={[ styles.navbar, { height: headerHeight, elevation: 4, zIndex: 4 }]}>
          <Animated.Image
            style={[
              styles.backgroundImage,
              { opacity: imageOpacity, transform: [{ translateY: imageTranslate }]}
            ]}
            source={{ uri: this.props.image }}
          />
          <Animated.View style={[ styles.bar, { height: headerHeight } ]}>
            <View style={styles.bar}>
              <Animated.Text
                style={[ styles.title, { fontSize: fontSizeTranslate, paddingLeft: padLeftTranslate } ]}
              >
                {this.props.title}
              </Animated.Text>
            </View>
          </Animated.View>
        </Animated.View>
        <View style={{ flex: 1, backgroundColor: PRIMARY_COLOR }}>
          <ScrollView
            style={{ flex: 1 }}
            scrollEventThrottle={16}
            onScroll={Animated.event(
              [ { nativeEvent: { contentOffset: { y: this.state.scrollY }}} ]
            )}
          >
            {this.renderContent()}
          </ScrollView>
        </View>
        <View style={styles.iconContainer}>
          <View style={styles.iconRadius}>
            {this.renderBackButton()}
          </View>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  navbar: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    backgroundColor: PRIMARY_COLOR,
    overflow: 'hidden'
  },
  bar: {
    height: 350,
    justifyContent: 'flex-end',
    marginBottom: 16
  },
  title: {
    backgroundColor: 'transparent',
    fontWeight: '400',
    fontSize: 22,
    color: 'hsl(0, 0%, 100%)'
  },
  navbarIcon: {
    width: 24,
    height: 24,
    backgroundColor: 'transparent'
  },
  iconContainer: {
    position: 'absolute',
    left: 0,
    // This basically means bottom: 6 since we can't use bottom values here
    top: STATUS_BAR_HEIGHT + NAVBAR_HEIGHT - (48 + 6),
    elevation: 4,
    zIndex: 25
  },
  iconRadius: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: 48,
    height: 48,
    borderRadius: 24
  },
  backgroundImage: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    width: null,
    height: HEADER_MAX_HEIGHT,
    resizeMode: 'cover'
  }
})

export default Detail
